BUILD.bazel

```diff
@@ -0,0 +1,333 @@
+load(
+    "//ign_bazel:build_defs.bzl",
+    "IGNITION_FEATURES",
+    "IGNITION_ROOT",
+    "IGNITION_VISIBILITY",
+    "cmake_configure_file",
+    "generate_include_header",
+    "ign_config_header",
+    "ign_export_header",
+)
+
+package(
+    default_visibility = IGNITION_VISIBILITY,
+    features = IGNITION_FEATURES,
+)
+
+licenses(["notice"])
+
+exports_files(["LICENSE"])
+
+PROJECT_NAME = "ignition-sensors"
+
+PROJECT_MAJOR = 4
+
+PROJECT_MINOR = 0
+
+PROJECT_PATCH = 0
+
+# Generates config.hh based on the version numbers in CMake code.
+ign_config_header(
+    name = "config",
+    src = "include/ignition/sensors/config.hh.in",
+    cmakelists = ["CMakeLists.txt"],
+    extra_defines = [
+        "IGN_SENSORS_PLUGIN_PATH=ign_sensors/",
+        "PROJECT_LIBRARY_TARGET_NAME=ignition-sensors",
+    ],
+    project_name = PROJECT_NAME,
+    project_version = (PROJECT_MAJOR, PROJECT_MINOR, PROJECT_PATCH),
+)
+
+sensors = [
+    "AirPressureSensor",
+    "AltimeterSensor",
+    "CameraSensor",
+    "DepthCameraSensor",
+    "GpuLidarSensor",
+    "ImuSensor",
+    "Lidar",
+    "LogicalCameraSensor",
+    "MagnetometerSensor",
+    "RenderingSensor",
+    "RgbdCameraSensor",
+    "ThermalCameraSensor",
+]
+
+# NOTE the below sensors names MUST correspond to the above sensors by index
+sensor_names = [
+    "air_pressure",
+    "altimeter",
+    "camera",
+    "depth_camera",
+    "gpu_lidar",
+    "imu",
+    "lidar",
+    "logical_camera",
+    "magnetometer",
+    "rendering",
+    "rgbd_camera",
+    "thermal_camera",
+]
+
+ign_export_header(
+    name = "include/ignition/sensors/Export.hh",
+    export_base = "IGNITION_SENSORS",
+    lib_name = "ignition-sensors",
+    visibility = ["//visibility:private"],
+)
+
+[
+    ign_export_header(
+        name = "include/ignition/sensors/%s/Export.hh" % sensor,
+        export_base = "IGNITION_SENSORS_%s" % sensor.upper(),
+        lib_name = "ignition-sensors",
+        visibility = ["//visibility:private"],
+    )
+    for sensor in sensor_names
+]
+
+sensor_sources = ["src/%s.cc" % sensor for sensor in sensors]
+
+sensor_headers = ["include/ignition/sensors/%s.hh" % sensor for sensor in sensors]
+
+public_headers_no_gen = [
+    "include/ignition/sensors/GaussianNoiseModel.hh",
+    "include/ignition/sensors/ImageGaussianNoiseModel.hh",
+    "include/ignition/sensors/ImageNoise.hh",
+    "include/ignition/sensors/Manager.hh",
+    "include/ignition/sensors/Noise.hh",
+    "include/ignition/sensors/RenderingEvents.hh",
+    "include/ignition/sensors/Sensor.hh",
+    "include/ignition/sensors/SensorFactory.hh",
+    "include/ignition/sensors/SensorTypes.hh",
+    "src/PointCloudUtil.hh",
+] + sensor_headers
+
+sources = [
+    "src/GaussianNoiseModel.cc",
+    "src/ImageGaussianNoiseModel.cc",
+    "src/ImageNoise.cc",
+    "src/Manager.cc",
+    "src/Noise.cc",
+    "src/PointCloudUtil.cc",
+    "src/RenderingSensor.cc",
```

Comment:
`RenderingSensor.cc` would cause duplicate symbol as it will also be picked up by the "rendering" sensor. My solution right now is adding `if sensor != "RenderingSensor"` to the list comprehensions below.